documentation/troubleshooting.md

```diff
@@ -28,3 +40,189 @@ Account: [myname@example.org]
 $ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=myname@example.org
 Clusterrolebinding "myname-cluster-admin-binding" created
 ```
+
+### Pod and volumes can be scheduled in different zones using default provisioned
+
+The default provisioner in GKE don't have the `volumeBindingMode: "WaitForFirstConsumer"` option that can result in bad
+scheduling behaviour.
+I uses files to create storageclass:
+- samples/gke-storage-standard-wait.yaml
+- samples/gke-storage-ssd-wait.yaml (if you have ssd disks)
+
+
+## Operator can't perform the Action
+
+If you ask to ScaleUp or add a new DC, or ask for more ressources, CassKop will ask Kubernetes to schedule as you
+requested.
+But sometimes it is not possible to achieve the change because of a lack of ressources (memory/cpus) or because
+constraints can't be satisfied (Kubernetes nodes with specific labels available...)
+
+CassKop make uses of the PodDisruptionBudget to prevent CassKop to make some change on the CassandraCluster that could
+make more than 1 Cassandra node at a time. 
+
+If you have a Pod stuck in **pending** state, then you have at least 1 Pod in Disruption, and the PDB object will
+prevent you to make changes on statefulset because that mean that you will have more than 1 Cassandra down at a time.
+
+```console
+$ k get poddisruptionbudgets
+NAME             MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
+cassandra-demo   N/A             1                 0                     12m
+```
+
+The Operator logs this line when there is disruption on the Cassandra cluster:
+
+```logs
+INFO[3037] Cluster has Disruption on Pods, we wait before applying any change to statefulset  cluster=cassandra-demo dc-rack=dc1-rack1
+```
+
+
+### Example: can't ScaleUp
+
+In this example I ask a ScaleUp but it can't perform :
+
+```console
+$ k get pods
+NAME                                                              READY   STATUS    RESTARTS   AGE
+cassandra-demo-dc1-rack1-0                                        1/1     Running   0          16h
+cassandra-demo-dc1-rack1-1                                        0/1     Pending   0          12m
+cassandra-demo-dc1-rack2-0                                        1/1     Running   0          16h
+cassandra-demo-dc1-rack3-0                                        1/1     Running   0          16h
+```
+
+the cassandra-demo-dc1-rack1-1 pod is Pending and can't be scheduled.
+
+If we looked at the pod status we will see this message :
+
+```
+Warning  FailedScheduling   51s (x17 over 14m)    default-scheduler   0/6 nodes are available: 4 Insufficient cpu, 4 node(s) didn't match node selector.
+```
+
+Kubernetes can't find any Pod with sufficiant cpu and matching kubernetes nodes labels we askde in the topology section.
+
+To fix this, we can either:
+- add more kubernetes nodes that will satisfied our requirements.
```

Comment:
We can also potentially reduce memory/cpu limits

Reply:
@allamand ^

Reply:
^^